rusk-wallet/src/wallet/file_service.rs

```diff
@@ -0,0 +1,341 @@
+        if !wallet_path.is_file() {
+            return Err(Error::WalletFileMissing);
+        }
+
+        // Load the wallet file
+        let mut bytes = std::fs::read(wallet_path)?;
+
+        match file_version {
+            DatFileVersion::Legacy => {
+                if bytes[1] == 0 && bytes[2] == 0 {
+                    bytes.drain(..3);
+                }
+
+                bytes = decrypt(&bytes, pwd)?;
+
+                let seed = bytes[..]
+                    .try_into()
+                    .map_err(|_| Error::WalletFileCorrupted)?;
+
+                Ok((seed, 1))
+            }
+            DatFileVersion::OldWalletCli((major, minor, _, _, _)) => {
+                bytes.drain(..5);
+
+                let content = decrypt(&bytes, pwd)?;
+                let buff = &content[..];
+
+                let seed =
+                    buff.try_into().map_err(|_| Error::WalletFileCorrupted)?;
+
+                match (major, minor) {
+                    (1, 0) => Ok((seed, 1)),
+                    (2, 0) => Ok((seed, buff[0])),
+                    _ => Err(Error::UnknownFileVersion(major, minor)),
+                }
+            }
+            DatFileVersion::RuskBinaryFileFormat(_) => {
+                let rest = bytes.get(12..(12 + 96));
+
+                if let Some(rest) = rest {
+                    let content = decrypt(rest, pwd)?;
+
+                    if let Some(seed_buf) = content.get(0..65) {
+                        let seed = seed_buf[0..64]
+                            .try_into()
+                            .map_err(|_| Error::WalletFileCorrupted)?;
+
+                        let addr_count = &seed_buf[64..65];
+
+                        Ok((seed, addr_count[0]))
+                    } else {
+                        Err(Error::WalletFileCorrupted)
+                    }
+                } else {
+                    Err(Error::WalletFileCorrupted)
+                }
+            }
+        }
+    }
+}
+
+/// Provides access to the wallet file path, profile directory and network name,
+/// and implements by default other useful methods
+pub trait WalletFilePath {
+    // Methods to implement ===================================================
+
+    /// Returns the path of the wallet file
+    fn wallet_path(&self) -> &PathBuf;
+    /// Returns the mutable path of the wallet file
+    fn wallet_path_mut(&mut self) -> &mut PathBuf;
+    /// Returns the directory of the profile
+    fn profile_dir(&self) -> &PathBuf;
+    /// Returns the network name for different cache locations
+    /// e.g, devnet, testnet, etc.
+    fn network(&self) -> Option<&String>;
+    /// Returns the mutable network name
+    fn network_mut(&mut self) -> &mut Option<String>;
+
+    // Automatically implemented methods =======================================
+
+    /// Sets the network name for different cache locations.
+    /// e.g, devnet, testnet, etc.
+    fn set_network_name(&mut self, network: Option<String>) {
```

Comment:
We're duplicating a lot of methods here, we have 2 traits that have duplicate methods why is that?

Reply:
We are dealing primarily with `Wallet` that has the `file: F` property. The user of the wallet doesn't have to know how the `file: F` is implemented and that it has the associated `PathBufferWrapper`. Hence I re-exported getters of the `PathBufferWrapper` in the `SecureWalletFile` trait to make the `file`'s API more user-friendly and enable user to call all necessary functions from the `file`.